shared/hooks/BackendAPI/checkForTokenExpiration.js

```diff
@@ -0,0 +1,12 @@
+import parseJWT from 'jwt-decode';
+import LuigiClient from '@luigi-project/client';
+
+export function checkForTokenExpiration(idToken) {
+  try {
+    const expirationTimestamp = parseJWT(idToken).exp * 1000;
+    if (new Date(expirationTimestamp) < Date.now()) {
+      // we cannot run parent.location.reload() here
+      LuigiClient.sendCustomMessage({ id: 'busola.refreshMainFrame' });
+    }
+  } catch (_) {} // ignore errors from non-JWT tokens
```

Comment:
Do we want swallow all exception here?

Reply:
Yeah, because idToken may be a non-jwt token, for example in case kubeconfig uploaded from file.